woocommerce/google-listings-and-ads
# Improved Default Issue Type and Issues Solved state

### Changes proposed in this Pull Request:



Partially Closes #1165 

This PR improves 2 things:

- Improves the Active Issue type logic as follow up for the comment here pcTzPl-da-p2#comment-430  and here https://github.com/woocommerce/google-listings-and-ads/pull/1176#discussion_r777647374

- Adds the Solved Issues state following the design direction in this Figma https://www.figma.com/file/eQ9O4m2flzrcAiDMXkOW0m/Google-Listings-%26-Ads-v1.x?node-id=4337%3A105188

### Screenshots:

![Screenshot 2022-01-14 at 19 12 47](https://user-images.githubusercontent.com/5908855/149566291-c569f14e-8309-448b-a445-59d92fc5ce85.png)




### Detailed test instructions:

1. Enter in GLA Product Feed section in an account with both Product and Account issues
2. See that Account Issues is selected by default and it shows the issues as normal
3. Switch to product Issues and see that it shows the Product issues as normal
4. Reload the page with Product issues selected. After reload. It should be Product Issues selected.
5. Delete in the database All your Account issues. You can use this command `DELETE FROM wp_gla_merchant_issues WHERE type = 'account';` 
6. Reload the page (still Product Issues selected) see that they load correctly. 
7. Switch to account issues. You should see a message similar to the Screenshot one. Saying that Account issues are solved and you should check Product issues tab. 
8. Refresh the page (Now Account issue is selected) after reload Account issues are selected. 
9. Delete from the URL the searchQuery `&issueType=account`. Enter to the page.
10. Product issues is selected by default since Account issues has no issues.
11. Delete your product issues.  `DELETE FROM wp_gla_merchant_issues WHERE type = 'product';` 
12. There is no Issues block anymore. And you see No issues to resolve 🎉 in the summary.



### Changelog entry

> Tweak - Informative message when here are no issues to solve
> Tweak - Issue type with issues is active by default
